bip-XXXX.mediawiki

```diff
@@ -74,13 +77,15 @@ This section is non-normative, but provides some examples of expected deployment
+In this setup, a Mining Controller (eg a cheap ARM-based server/or even arduino/or cpu-less controllers) accepts incoming connections from ASIC miners. These connections can either use stratum (for backward compatibility) or the new work protocol (optionally in headers mode, which greatly simplifies the logic at the ASIC level - allowing the ASIC firmware to skip even implementing merkle-path-hashing/coinbase transaction construction logic). This Mining Controller gets work via the work protocol (never in headers mode) from a local bitcoind (optionally on the same device, for simplicity) and connects to one or more remote pools using the pool protocol. With a BLOCK_TEMPLATE from the bitcoind and a PAYOUT_INFO from the pool server, the Mining Controller can construct unique work which pays out to the pool but contains transactions selected by a local node for distribution to the ASICs. Blocks which meet the pool's SHARE_DIFFICULTY's ''share_target'' are sent (with just the coinbase transaction, merkle path, and block header) to the pool for verification and payout, and blocks which meet the Bitcoin network difficulty are sent to the local bitcoind. The pool can optionally request that full blocks be sent at a secondary difficulty (using a weak-block-based compression scheme) so that the pool receives full-difficulty blocks for relaying as well.
 
 ===Simple but Efficient Pool Architecture===
 
 TODO: Something about how having only one pool server is great cause you can multiplex multiple clients over the pool protocol so you can have a bunch of stratum proxies + bitcoinds globally for effeciency and then only have one (or two, for redundancy/GFW issues) pool server centrally/securely hosted, greatly simplifying management and security of a pool.
 
 ==Specification of Work Protocol==
+(suggestion message_type/message_length/8-64byte control or signature - makes simpler protocol parsing and lower attack vectors
+or even message_length/8-64byte control or signature and then message_type makes parser logics simpler.
```

Comment:
Added a 4 byte magic value but only for broadcast mode - TCP provides similar results implicitly already so its just redundant in TCP mode. See number 3 at https://github.com/TheBlueMatt/bips/blob/betterhash/bip-XXXX.mediawiki#Final_Work_Protocol_over_Broadcast_Mediums